### ISSUE TITLE
NoDestroyUndo component has no effect.

### ISSUE THREAD

**Issue by mpmxyz** [Yellow-Dog-Man/Resonite-Issues#979]
### Describe the bug?

```NoDestroyUndo``` does not implement ```INoDestroyUndo``` and is therefore ignored when checking if a destroyed object may be revived using undo context actions.

### To Reproduce

1. Create an object
2. Attach "Transform/Tagging/NoDestroyUndo" to the root of the object.
3. Destroy object either via grab-context menu or using the DevTool/Inspector
4. Check if you can undo destroying the object via context menu

### Expected behavior

The object should be destroyed and there should be no context menu option to undo that.

### Resonite Version Number

2023.12.21.1127

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Desktop

### Log Files

[ANONYMOUS - 2023.12.21.1127 - 2023-12-22 22_32_18.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/13756871/ANONYMOUS.-.2023.12.21.1127.-.2023-12-22.22_32_18.log)

**Comment by mpmxyz** [comment]
A bit of trivia:
I created and tested an object with this component today, only to be confused by the results. (no effect)
So, I collected information for a bug report and at the end found out _I already did this_. lol

TLDR: Bug confirmed for version 2026.9.18.82.
PS: I won't judge you but wouldn't it be more appropriate as an XS or S sized task with <1 dog walks considering it (likely) only needs adding the marker interface `INoDestroyUndo` on top of the base class?
